lib/symboldatabase.cpp

```diff
@@ -4202,8 +4204,41 @@ static const Token* skipPointersAndQualifiers(const Token* tok)
     return tok;
 }
 
+const Token* skipTypeOrFunctionIdentifier(const Token* tok)
+{
+    if (tok == nullptr)
+        return nullptr;
+    while (tok->next() != nullptr && (tok->next()->str() == "::" || tok->next()->str() == "<"))
```

Comment:
I don't think %name% will work because this is a while not an if and there can be consecutive non-names while getting to the end of the type for example the patten ```>::``` in ```A<B>::C```.

Otherwise I agree with you that the Token::Match expression is a much cleaner expression of intent for a not particularly onerous performance hit so I've made the change without %name%.